bagit.py

```diff
@@ -632,6 +645,23 @@ def _validate_bagittxt(self):
             if first_line.startswith(codecs.BOM_UTF8):
                 raise BagValidationError("bagit.txt must not contain a byte-order mark")
 
+    def _path_is_dangerous(self, path):
+        """
+        Return true if path looks dangerous, i.e. potentially operates
+        outside the bagging directory structure, e.g. ~/.bashrc, ../../../secrets.json,
+            \\?\c:\, D:\sys32\cmd.exe
+        """
+        # Disallow dots, slashes, and tildes as first character
+        disallow = r'.\/~'
+        if path[0] in disallow:
+            return True
+        # Disallow drive letters at beginning of path, e.g c:, D:
+        drive_pat = re.compile(r'^[A-Za-z]:')
+        if drive_pat.match(path):
```

Comment:
I would probably remove the separate compile step and just write this as `re.match(…)` since the`re` module caches both calls to `re.compile` and the module-level `match`, `sub`, etc. functions. It's possible that in a large program this might exceed the `re` cache size and in that case we could see a modest win by making a module-level `DRIVE_PATTERN_RE` compiled constant and using that but I'd want to see a benchmark before not going with the simplest code